WeatherApp-UIKit/WeatherApp-UIKit/SceneDelegate.swift

```diff
@@ -21,8 +21,8 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
 
     let rootVC = UIStoryboard(name: "Main", bundle: nil)
       .instantiateInitialViewController() as! ViewController
-    rootVC.viewModel = WeatherViewModel(currentLocation: LocationProvider.shared.currentLocation)
-
+    rootVC.viewModel = stateMachineViewModel
+    
     window.rootViewController = rootVC
```

Comment:
This commit should be the last one before you actually swap the implementations.